gcloud-java-core/src/main/java/com/google/gcloud/BaseServiceException.java

```diff
@@ -16,26 +16,115 @@
 
 package com.google.gcloud;
 
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.Collections;
+import java.util.Set;
+
 /**
  * Base class for all service exceptions.
  */
 public class BaseServiceException extends RuntimeException {
 
-  private static final long serialVersionUID = 5028833760039966178L;
+  protected static class Error {
+
+    private final Integer code;
+    private final String reason;
+
+    public Error(Integer code, String reason) {
+      this.code = code;
```

Comment:
We use `null` to mean "any" in the `retriablyErrors()` `Set`. We could use `UNKNOWN_CODE` if you prefer.